mindmeld/augmentation.py

```diff
@@ -0,0 +1,315 @@
+        return augmented_queries
+
+
+class MultiLingualParaphraser(Augmentor):
+    """Paraphraser class for generating paraphrases based on language code of the app
+    (currently supports: French, Italian, Spanish, Portuguese, and Romanian).
+    """
+
+    def __init__(self, lang, paths, path_suffix, params, resource_loader):
+        """Initializes a multi-lingual paraphraser.
+
+        Args:
+            lang (str): The lang code for paraphrasing
+            paths (list): Path rules for fetching relevant files to Paraphrase.
+            path_suffix (str): Suffix to be added to new augmented files.
+            params (dict): Model params.
+            resource_loader (object): Resource Loader object for the application.
+        """
+        super().__init__(
+            lang=lang,
+            paths=paths,
+            path_suffix=path_suffix,
+            resource_loader=resource_loader,
+        )
+
+        self.torch_device = (
+            "cuda" if _get_module_or_attr("torch.cuda", "is_available")() else "cpu"
+        )
+
+        MarianTokenizer = _get_module_or_attr("transformers", "MarianTokenizer")
+        MarianMTModel = _get_module_or_attr("transformers", "MarianMTModel")
+
+        en_model_name = "Helsinki-NLP/opus-mt-ROMANCE-en"
+        self.en_tokenizer = MarianTokenizer.from_pretrained(en_model_name)
+        self.en_model = MarianMTModel.from_pretrained(en_model_name)
+        self.en_model.to(self.torch_device)
+
+        target_model_name = "Helsinki-NLP/opus-mt-en-ROMANCE"
+        self.target_tokenizer = MarianTokenizer.from_pretrained(target_model_name)
+        self.target_model = MarianMTModel.from_pretrained(target_model_name).to(
+            self.torch_device
+        )
+
+        # Use default model params if not customized for app
+        self.params = params or {
+            "num_beams": 3,
+            "num_return_sequences": 3,
+            "top_k": 0,
+            "temperature": 1.0,
+        }
+
+    def _translate(self, *, template, queries, model, tokenizer, **kwargs):
+        """The core translation step for forward and back translation.
+
+        Args:
+            template (lambda func): Structure input text to model.
+            queries (list(str)): List of input queries.
+            model: Machine translation model (en-ROMANCE or ROMANCE-en).
+            tokenizer: Language tokenizer for input query text.
+        """
+        queries = [template(query) for query in queries]
+        encoded = tokenizer.prepare_seq2seq_batch(queries, return_tensors="pt")
+        for key in encoded:
+            encoded[key] = encoded[key].to(self.torch_device)
+
+        translated = model.generate(**encoded, **kwargs)
+        translated_queries = tokenizer.batch_decode(
+            translated, skip_special_tokens=True
+        )
+        return translated_queries
+
+    def augment_queries(self, queries):
+        translated_queries = self._translate(
```

Comment:
Shouldn't we set a batch size here and then call translate()?

Reply:
Created batching within translate.